src/utils/third-party-templates.ts

```diff
@@ -0,0 +1,44 @@
+import { RawOptions } from "../types";
+
+// Gets the data from the argument passed to the `--template` option.
+// e.g. owner/project:branch => { githubBranchUrl, githubUrl, branch, owner, project }
+export const getDataFromTemplateArgument = (template: string) => {
+  // Check format: owner/project:branch (branch is optional)
+  const regex = /^[^/]+\/[^/]+(:[^/]+)?$/;
+  if (!regex.test(template)) {
+    throw new Error(
+      `Invalid template format. Use "owner/project" or "owner/project:branch"`
+    );
+  }
+
+  // Extract owner, project and branch
+  const owner = template.split("/")[0];
+  const project = template.split(":")[0].split("/")[1];
+  const branch = template.split(":")[1];
+
+  // Check if the repo exists.
```

Comment:
where is the check?

Reply:
Definitely now here! haha Had there originally but extracted it away.

4c08de70111cdd564283e014ebb95b35754efc29